commercial-paper/commercial-paper-app/src/main/java/com/effective/hlf/hlf/commercialpaper/CommercialPaperContractStub.java

```diff
@@ -32,4 +32,11 @@ public CommercialPaper redeem(String... args) throws ContractException, Interrup
         byte[] response = contract.submitTransaction("redeem", args);
         return CommercialPaper.deserialize(response);
     }
+
+    public CommercialPaper read(String... args) throws ContractException, InterruptedException, TimeoutException {
+        byte[] response = contract.submitTransaction("read", args);
```

Comment:
You submit here but in contract you have @Transaction(submit=false) - looks contradictive and suspicious. 

For read transactions we should use _contract.evaluateTransaction(...)_ instead.